### PR TITLE
Add section on pack dependencies

### PR DESCRIPTION
**Pull request by blag**
Documentation for https://github.com/StackStorm/st2/pull/4769, which adds support for specifying pack dependencies to `pack.yaml`.

### CODE REVIEW

File: docs/source/packs.rst [hunk 1]
```diff
@@ -133,6 +133,35 @@ version or **upgrade to latest** if the version is not specified. Your config fi
...
 we recommend to always specify versions in case there are major changes in ``latest``.
 
+Pack Dependencies
+~~~~~~~~~~~~~~~~~
+
+.. note::
+
+    New feature! As of StackStorm 3.2.
+
+If you pack uses actions from other packs, you can specify them in the ``dependencies`` section of
```
Comment (punkrokk) [inline docs/source/packs.rst]: `If your pack`

Comment (blag) [reply docs/source/packs.rst]: Fixed, thanks!

File: docs/source/packs.rst [hunk 2]
```diff
@@ -133,6 +133,35 @@ version or **upgrade to latest** if the version is not specified. Your config fi
...
+
+If you have dependency conflicts, the ``st2 pack install`` subcommand may error out, without
+installing any packs. If you would like to forcibly install a pack without installing its
+dependencies, you can use the ``--skip-dependencies`` flag:
+
+.. code-block:: bash
+
+    st2 pack install --skip-dependencies my-custom-pack
+
 To uninstall a pack, use ``remove``:
```
Comment (armab) [inline docs/source/packs.rst]: This is part of `Installing a pack` section.
Looks like the new `Pack Dependencies` section breaks it into 2 pieces.

Would be better to move `Pack Dependencies` after pack uninstall instructions.

Comment (m4dcoder) [reply docs/source/packs.rst]: I think it makes sense to be part of or after pack install section.  Otherwise, folks will just miss this.

Comment (blag) [reply docs/source/packs.rst]: Yeah, if anything, we should move the pack uninstallation instructions out of the *installation* section. This is like the old *Start* menu -> Shutdown option in Windows.

But that is outside the scope of this PR.

Comment (armab) [reply docs/source/packs.rst]: With the way how `Pack Dependencies` was added, there is no separation and so Uninstall is part of Dependencies header/section.

I'm good if we place this under the new Uninstall section.

Comment (blag) [reply docs/source/packs.rst]: Pack dependencies only relate to installing a pack though. If I move the subsection below the uninstallation instructions, I think people will miss the dependency section entirely.

Comment (armab) [reply docs/source/packs.rst]: `Uninstall` can be somewhat related to `Install` section where removal of a pack comes as a conclusion. But it's not a part of `Pack Dependencies` which primary topic is special case about the dependencies syntax/functionality.

To fix this or we move that small pack removal note back to Install or add additional `Uninstall` sub-header. I'm good with any of these.

Comment (blag) [reply docs/source/packs.rst]: Fixed in https://github.com/StackStorm/st2docs/pull/936.

**Pull request merged**